File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
// sort array 
//check if num in one index is same as num in the next array 
//return true if yes 
// iterate completely through the arr and return false if not 

var containsDuplicate = function(nums) {
    nums.sort()
    
  for(let i = 0; i < nums.length; i++){
      if(nums[i] === nums[i + 1]){
          return true; 
      }
  }
    return false; 
};


//tc = o(n)
//sc = o(1)

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        Input:type nums: List[int]
       Output:rtype: bool
        
        We need to determine if there are any duplicates in our input. 
        
        create a dictionary to keep track of the frequency of numbers 
            loop through the nums list 
                if the curr num is present in the diction 
                     it is? then, we have a duplicate. Return true. 
                else if it isn't, then we add it and assign a freq of 1.      
            If none meet the duplicate check condition, then we have no duplicate. We return true. 
        """
        
        hashmap = {}
        for num in nums:
            if num in hashmap:
               return True
        
            else:
                hashmap[num] = 1 
                
        return False 
            